test/joseph/JosephAutoRebalance.t.sol

```diff
@@ -0,0 +1,1692 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.16;
+
+import "forge-std/Test.sol";
+import "forge-std/console2.sol";
+import "../TestCommons.sol";
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import {DataUtils} from "../utils/DataUtils.sol";
+import "../../contracts/libraries/math/IporMath.sol";
+import "../../contracts/libraries/Constants.sol";
+import "../../contracts/itf/ItfIporOracle.sol";
+import "../../contracts/itf/ItfMiltonUsdt.sol";
+import "../../contracts/itf/ItfMiltonUsdc.sol";
+import "../../contracts/itf/ItfMiltonDai.sol";
+import "../../contracts/itf/ItfJosephUsdt.sol";
+import "../../contracts/itf/ItfJosephUsdc.sol";
+import "../../contracts/itf/ItfJosephDai.sol";
+import "../../contracts/tokens/IpToken.sol";
+import "../../contracts/mocks/stanley/MockCase0Stanley.sol";
+import "../../contracts/mocks/spread/MockSpreadModel.sol";
+import "../../contracts/mocks/tokens/MockTestnetTokenDai.sol";
+import "../../contracts/mocks/tokens/MockTestnetTokenUsdc.sol";
+import "../../contracts/mocks/tokens/MockTestnetTokenUsdt.sol";
+
+contract JosephAutoRebalance is Test, TestCommons, DataUtils {
+    MockSpreadModel internal _miltonSpreadModel;
+    ItfIporOracle private _iporOracle;
+
+    MockTestnetTokenUsdt private _usdt;
+    IpToken private _ipTokenUsdt;
+
+    ItfMiltonUsdt private _itfMiltonUsdt;
+    ItfJosephUsdt private _itfJosephUsdt;
+    MiltonStorage private _miltonStorageUsdt;
+    MockCase0Stanley private _stanleyUsdt;
+
+    MockTestnetTokenDai private _dai;
+    IpToken private _ipTokenDai;
+    ItfMiltonDai private _itfMiltonDai;
+    ItfJosephDai private _itfJosephDai;
+    MiltonStorage private _miltonStorageDai;
+    MockCase0Stanley private _stanleyDai;
+
+    address internal _admin;
+    address internal _userOne;
+
+    function setUp() public {
+        _admin = address(this);
+        _userOne = _getUserAddress(1);
+
+        _usdt = getTokenUsdt();
+        _dai = getTokenDai();
+
+        _miltonSpreadModel = prepareMockSpreadModel(0, 0, 0, 0);
+
+        address[] memory tokenAddresses = new address[](2);
+        tokenAddresses[0] = address(_usdt);
+        tokenAddresses[1] = address(_dai);
+
+        _iporOracle = getIporOracleForManyAssets(_admin, _userOne, tokenAddresses, 1, 1, 1);
+    }
+
+    function testProvideLiquidityAndRebalanceUsdtCase01() public {
+        //given
+        _clearAndSetupSmartContractsUsdt();
```

Comment:
This method is confusing, I don't understand what `clear` mean.

Reply:
I changed to _setupSmartContractsUsdt